### PR TITLE
[documentation] Add health-check extension contract to observability docs

### LINKED ISSUE #592: [documentation] Add health-check extension contract to observability docs
## Why / Context
Health-check changes are frequent; explicit extension docs reduce onboarding time and implementation mistakes for future Codex tasks.

## Scope
**In scope**
- Document the health-check registration flow and expected result shape.
- Add a concise extension checklist for adding new checks.
- Cross-link observability and monitoring playbook docs.

**Out of scope**
- Rewriting the full monitoring architecture docs.
- Changing runtime health-check behavior.

Constraints:
- Keep changes small/mergeable.
- Deterministic tests only.
- Avoid touching unrelated generated artifacts unless required.

## Acceptance Criteria (required)
- [ ] `docs/OBSERVABILITY.md` includes a health-check extension contract section.
- [ ] Docs include at least one concrete file pointer for registration entrypoints.
- [ ] Monitoring playbook links to the contract section for triage context.
- [ ] Documentation changes pass existing docs/CI checks.

## Implementation Notes / Pointers
**Likely files / modules:**
- `docs/OBSERVABILITY.md`
- `docs/MONITORING_PLAYBOOK.md`
- `src/gpt_trader/monitoring/health_checks.py`

**Related tests:**
- `scripts/maintenance/docs_link_audit.py`
- `tests/unit/scripts/test_docs_maintenance.py`

**Edge cases to handle:**
- Avoid stale links when headings are updated.
- Keep examples aligned with current module naming.
- Ensure docs stay concise and actionable for agent-driven work.

## Commands (local)
- `make fmt`
- `make lint`
- `uv run pytest -q tests/unit/scripts/test_docs_maintenance.py`

## PR Requirements
- PR title should match the issue.
- PR body must include: `Fixes #<issue-number>`
- CI must be green; if Agent Artifacts Freshness fails, run `uv run agent-regenerate`, then commit `var/agents/...` updates.

## Codex-Ready Checklist (for the issue creator)
- [x] Clear acceptance criteria
- [x] At least one file pointer
- [x] Commands included
- [x] No ambiguous "do the right thing" language

### PR DESCRIPTION
**Pull request by Solders-Girdles**
Fixes #592

(Automated PR created by OpenClaw PR finisher)
Branch: codex/592-documentation-add-health-check-extension
Commit: f6db16217d3d2d94c6a5d7f36748c82ab09f7009

### CODE REVIEW

File: docs/OBSERVABILITY.md [hunk 1]
```diff
@@ -20,6 +20,45 @@ This page is intentionally thin. Metric names/labels and event schemas are gener
...
+      "<name>": {
+        "status": "pass" | "fail",
+        "details": { "...": "..." }
+      }
+    }
+  }
+  ```
+- Include `details["severity"]` as `warning` or `critical` when a failure should
+  degrade or fail `/health`. The health server also appends `performance` and
+  `signals` summaries to the `checks` payload.
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply docs/OBSERVABILITY.md]: Good catch — updated the docs to match the actual `/health` payload shape: `signals` is a top-level object (alongside `checks`), while the performance summary lives under `checks.performance`. Thanks for flagging.
